### ISSUE TITLE
Trimming wrapper from Your react tutorial

### ISSUE THREAD

**Issue by piernik** [ckeditor/ckeditor5#7826]
You provided great tutorial: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/using-react-in-a-widget.html
I used it for my own purposes but have one issue.
Here it creates wrapper:
```javascript
conversion.for( 'editingDowncast' ).elementToElement( {
            model: 'productPreview',
            view: ( modelElement, viewWriter ) => {
                // In the editing view, the model <productPreview> corresponds to:
                //
                // <section class="product" data-id="...">
                //     <div class="product__react-wrapper">
                //         <ProductPreview /> (React component)
                //     </div>
                // </section>
                const id = modelElement.getAttribute( 'id' );

                // The outermost <section class="product" data-id="..."></section> element.
                const section = viewWriter.createContainerElement( 'section', {
                    class: 'product',
                    'data-id': id
                } );

                // The inner <div class="product__react-wrapper"></div> element.
                // This element will host a React <ProductPreview /> component.
                const reactWrapper = viewWriter.createRawElement( 'div', {
                    class: 'product__react-wrapper'
                }, function( domElement ) {
                    // This the place where React renders the actual product preview hosted
                    // by a UIElement in the view. You are using a function (renderer) passed as
                    // editor.config.products#productRenderer.
                    renderProduct( id, domElement );
                } );

                viewWriter.insert( viewWriter.createPositionAt( section, 0 ), reactWrapper );

                return toWidget( section, viewWriter, { label: 'product preview widget' } );
            }
        } );
```
In my example I don't need wrapper, but I do want to use `renderProduct, that uses `domElement`.
But `createContainerElement` does not have function to place render function and `toWidget` needs `containerElement` as first argument.
Am I able to trim `wrapper` ?

**Comment by Mgsy** [comment]
@oleq, can you take a look at it?

**Comment by oleq** [comment]
> In my example I don't need wrapper, but I do want to use `renderProduct, that uses`domElement`. But`createContainerElement`does not have function to place render function and`toWidget`needs`containerElement`as first argument. Am I able to trim`wrapper\` ?

You're right, `createContainerElement()` does not allow custom rendering functions because [`ContainerElement` must be 100% under the control of the editor editing engine](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/editing-engine.html#element-types-and-custom-data). The only place where you get access to the native DOM is inside a `RawElement` or `UIElement`. 

However, neither of them can be "widgetized" (via `toWidget()`) straight away without a wrapper because (long story short) making a widget out of element means injecting some editor internals into this element and since the editor has no control over what happens inside of it (your rendering function can do anything, e.g. mess with editor internals), this is simply a no go.

TL;DR: This container element wrapper is mandatory if you create a widget and there's no way to get around it. Sorry.
